### ISSUE TITLE
Regression from 2.1 to 2.2: markdownlint-cli no longer detected

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [webpro/knip#108]
In [template-typescript-node-package](https://github.com/JoshuaKGoldberg/template-typescript-node-package), bumping `knip` from 2.1.3 to 2.2.0 introduces a new complaint:

```plaintext
$ knip

Unused devDependencies (1)
markdownlint-cli  package.json
```

If we remove that package and run `pnpm lint:md` (`markdownlint`) again, the markdownlint CLI isn't found (because it's provided by `markdownlint-cli`):

```plaintext
sh: markdownlint: command not found
```

**Comment by navin-moorthy** [comment]
I second that

<img width="404" alt="image" src="https://user-images.githubusercontent.com/39694575/228912999-884f9044-572d-4bc5-bf2b-3c52b2e3c0cd.png">

**Comment by webpro** [comment]
Darn, this must have happened after a large refactoring recently which I was actually happy about. Not sure how this slipped through, because there already were similar tests (e.g. for `tsc` being a binary of `typescript`). Anyway, now binaries get the `bin:` prefix which makes things more clear, both implementation and output wise. More tests have been added as well. Until now, binaries could be disguised as package names, or vice versa, but that's no longer the case.

Considering these are bug fixes, so here's [v2.3.1](https://github.com/webpro/knip/releases/tag/2.3.1).

**Comment by webpro** [comment]
A further improvement is in [v2.3.2](https://github.com/webpro/knip/releases/tag/2.3.2) that should wrap things up for now :)
